### PR TITLE
executor_test: clean enviroment for each test.

### PR DESCRIPTION
**Pull request by hanfei1991**
We should clean the evironment for every test, so as to avoid one test can use a table that created by another test.
@shenli @coocood @ngaut @XuHuaiyu @zimulala PTAL

### CODE REVIEW

File: executor/aggregate_test.go [hunk 1]
```diff
@@ -259,6 +259,7 @@ func (s *testSuite) TestAggregation(c *C) {
 	tk.MustExec("insert into t2 values(22, 2), (3, 12), (38, 98)")
 	result = tk.MustQuery("SELECT COALESCE ( + 1, cor0.col0 ) + - CAST( NULL AS DECIMAL ) FROM t2, t1 AS cor0, t2 AS cor1 GROUP BY cor0.col1")
 	result.Check(testkit.Rows("<nil>", "<nil>"))
+	tk.MustExec("drop table t, t1, t2")
```
Comment (ngaut) [inline executor/aggregate_test.go]: Could you use defer statement?

Comment (hanfei1991) [inline executor/aggregate_test.go]: any benifit of using defer?

Comment (ngaut) [inline executor/aggregate_test.go]: Clean, and you will never forget to clean up table.

**Comment by hanfei1991** [comment]
@ngaut PTAL

File: executor/aggregate_test.go [hunk 2]
```diff
@@ -59,8 +59,11 @@ func (m *MockExec) Close() error {
 }
 
 func (s *testSuite) TestAggregation(c *C) {
-	defer testleak.AfterTest(c)()
 	tk := testkit.NewTestKit(c, s.store)
+	defer func() {
+		testleak.AfterTest(c)()
+		tk.MustExec("drop table if exists t, t1, t2")
```
Comment (coocood) [inline executor/aggregate_test.go]: This should be executed before `testleak.AfterTest(c)()`

File: executor/executor_test.go [hunk 3]
```diff
@@ -632,10 +661,32 @@ func (s *testSuite) TestJoin(c *C) {
 }
 
 func (s *testSuite) TestMultiJoin(c *C) {
-	defer testleak.AfterTest(c)()
 	tk := testkit.NewTestKit(c, s.store)
+	defer func() {
+		testleak.AfterTest(c)()
+		tk.MustExec("drop table if exists t15")
```
Comment (zimulala) [inline executor/executor_test.go]: These statements can be synthesized into a statement.

**Comment by coocood** [comment]
Usually we call defer right after resource is created, in this case, the pattern would be.

```
tk.MustExec("create table t (a int)")
defer tk.MustExec("drop table if exists t")
```

So it's very easy to check that resource has been cleaned up.

**Comment by zimulala** [comment]
LGTM

**Comment by shenli** [comment]
@hanfei1991 Please add description for the PR.

**Comment by hanfei1991** [comment]
@ngaut @coocood PTAL

File: executor/aggregate_test.go [hunk 4]
```diff
@@ -59,8 +59,11 @@ func (m *MockExec) Close() error {
 }
 
 func (s *testSuite) TestAggregation(c *C) {
-	defer testleak.AfterTest(c)()
 	tk := testkit.NewTestKit(c, s.store)
+	defer func() {
```
Comment (tiancaiamao) [inline executor/aggregate_test.go]: I can't understand this change

File: executor/executor_ddl_test.go [hunk 5]
```diff
@@ -22,8 +22,11 @@ import (
 )
 
 func (s *testSuite) TestTruncateTable(c *C) {
-	defer testleak.AfterTest(c)()
 	tk := testkit.NewTestKit(c, s.store)
+	defer func() {
+		defer testleak.AfterTest(c)()
```
Comment (tiancaiamao) [inline executor/executor_ddl_test.go]: ditto

File: executor/executor_ddl_test.go [hunk 6]
```diff
@@ -36,8 +39,11 @@ func (s *testSuite) TestTruncateTable(c *C) {
 }
 
 func (s *testSuite) TestCreateTable(c *C) {
-	defer testleak.AfterTest(c)()
 	tk := testkit.NewTestKit(c, s.store)
+	defer func() {
+		defer testleak.AfterTest(c)()
```
Comment (tiancaiamao) [inline executor/executor_ddl_test.go]: ditto

rest LGTM

**Comment by tiancaiamao** [comment]
LGTM

There is one concern that if we want run unit test parallel, `cleanEnv` would make trouble.
PTAL @coocood

**Comment by coocood** [comment]
LGTM

**Pull request merged**